community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactions.java

```diff
@@ -166,38 +143,16 @@ protected void dispose( KernelTransactionImplementation tx )
      */
     public List<KernelTransaction> activeTransactions()
     {
-        List<KernelTransaction> output = new ArrayList<>();
-        for ( KernelTransactionImplementation tx : allTransactions )
-        {
-            if ( tx.isOpen() )
-            {
-                output.add( tx );
-            }
-        }
-
-        return output;
+        return allTransactions.stream().filter( KernelTransaction::isOpen ).collect( toList() );
```

Comment:
Transaction is open when it is not closing and not closed. `KernelTransactionImplementation` has two boolean flags inside `closed` and `closing`. Second one is basically used to not allow concurrent closing of a transaction. This is a bit strange. I'll investigate the possibility of simplification.